### PR TITLE
Javamop spec

### PR DESCRIPTION
**Pull request by yzhang90**

### CODE REVIEW

File: monitor/getConsistency.mop [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+
+    unsynced event getResCall after (byte[] key, int rId) returning (byte[] result):
+        call(byte[] HaloDBStorageEngine.get(byte[], int, int, int))
+        && args(key, rId, ..) {
+            long keyLong = MopUtils.bytesToLong(key);
+            PutResponseWithPending prwp = snapshotAtGetReq.get(rId);
+            if (result == null) {
+                if (prwp.lastSuccessfulPut != null) {
```
Comment (msaxena2) [inline monitor/getConsistency.mop:139]: I'm not convinced this is correct. The response is still valid if there's a delete in the pendingSet right?

File: monitor/dependencies/MonitorRecord.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        this.key = key;
+        this.checksum = checksum;
+        this.seq = seq;
+    }
+
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
```
Comment (msaxena2) [inline monitor/dependencies/MonitorRecord.java:18]: Is this hashing scheme safe (as in doesn't lead to collisions)?

Comment (yzhang90) [reply monitor/dependencies/MonitorRecord.java:18]: i think this is the default hash code implementation generated by Eclipse

File: monitor/getConsistency.mop [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.zip.CRC32;
+import com.oath.halodb.*;
+import com.oath.halodb.javamop.*;
+
+getConsistency() {
+
+    Map<Integer, PutResponseWithPending> snapshotAtGetReq = new ConcurrentHashMap<Integer, PutResponseWithPending>();
```
[the remark this answers is not in the document]
Comment (yzhang90) [reply monitor/getConsistency.mop:12]: Do you mean that we should use Pair to wrap putResponse and Pending set instead of creating a new class?

**Pull request merged**